Apply patch: use calendar.timegm instead of manual broken timezone math.

Author: marienz
Review: itamar
Fixes: #5583

Index: /trunk/twisted/test/test_log.py
===================================================================
--- /trunk/twisted/test/test_log.py	(revision 32453)
+++ /trunk/twisted/test/test_log.py	(revision 34145)
@@ -6,5 +6,5 @@
 """
 
-import os, sys, time, logging, warnings
+import os, sys, time, logging, warnings, calendar
 from cStringIO import StringIO
 
@@ -392,7 +392,5 @@
         human-readable string.
         """
-        # There is no function in the time module which converts a UTC time
-        # tuple to a timestamp.
-        when = time.mktime((2001, 2, 3, 4, 5, 6, 7, 8, 0)) - time.timezone
+        when = calendar.timegm((2001, 2, 3, 4, 5, 6, 7, 8, 0))
 
         # Pretend to be in US/Eastern for a moment
